diff -Nuar zfs-zfs-0.6.2.orig/config/kernel-bdi-setup-and-register.m4 zfs-zfs-0.6.2/config/kernel-bdi-setup-and-register.m4
--- zfs-zfs-0.6.2.orig/config/kernel-bdi-setup-and-register.m4	2014-02-21 22:28:36.320008551 +0100
+++ zfs-zfs-0.6.2/config/kernel-bdi-setup-and-register.m4	2014-02-21 22:41:44.896908777 +0100
@@ -1,12 +1,14 @@
 dnl #
 dnl # 2.6.34 API change
-dnl # The bdi_setup_and_register() helper function is avilable and
+dnl # The bdi_setup_and_register() helper function is available and
 dnl # exported by the kernel.  This is a trivial helper function but
 dnl # using it significantly simplifies the code surrounding setting
 dnl # up and tearing down the bdi structure.
 dnl #
 AC_DEFUN([ZFS_AC_KERNEL_BDI_SETUP_AND_REGISTER],
 	[AC_MSG_CHECKING([whether bdi_setup_and_register() is available])
+	tmp_flags="$EXTRA_KCFLAGS"
+	EXTRA_KCFLAGS="-Wno-unused-result"
 	ZFS_LINUX_TRY_COMPILE_SYMBOL([
 		#include <linux/backing-dev.h>
 	], [
@@ -18,4 +20,5 @@
 	], [
 		AC_MSG_RESULT(no)
 	])
+	EXTRA_KCFLAGS="$tmp_flags"
 ])
